### PR TITLE
feat: upd wasmd to v0.51

### PR DESCRIPTION
**Pull request by swelf19**

### CODE REVIEW

**Review by pr0n00gler — APPROVED** [review]

**Comment by swelf19** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/8941365683

**Review by pr0n00gler — CHANGES_REQUESTED** [review]
Static binary build doesn't work:

```
$ make build-static-linux-amd64
...
...
------
 > [builder 8/8] RUN --mount=type=cache,target=/root/.cache/go-build     --mount=type=cache,target=/root/go/pkg/mod     go build       -mod=readonly       -tags "netgo,ledger,muslc"       -ldflags "-X github.com/cosmos/cosmos-sdk/version.Name="neutron"               -X github.com/cosmos/cosmos-sdk/version.AppName="neutrond"               -X github.com/cosmos/cosmos-sdk/version.Version=3.0.2-178-g25407263               -X github.com/cosmos/cosmos-sdk/version.Commit=254072639b370917d709d189cd936871279075d4               -X github.com/cosmos/cosmos-sdk/version.BuildTags='netgo,ledger'               -X github.com/neutron-org/neutron/app.EnableSpecificProposals=${ENABLED_PROPOSALS}               -w -s -linkmode=external -extldflags '-Wl,-z,muldefs -static'"       -trimpath       -o /neutron/build/neutrond       /neutron/cmd/neutrond:
4.626 # github.com/neutron-org/neutron/v4/cmd/neutrond
4.626 /usr/local/go/pkg/tool/linux_amd64/link: running gcc failed: exit status 1
4.626 /usr/lib/gcc/x86_64-alpine-linux-musl/12.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: cannot find -lwasmvm_muslc.x86_64: No such file or directory
4.626 collect2: error: ld returned 1 exit status
4.626
------
Dockerfile.builder:41
--------------------
  40 |     # Build neutrond binary
  41 | >>> RUN --mount=type=cache,target=/root/.cache/go-build \
  42 | >>>     --mount=type=cache,target=/root/go/pkg/mod \
  43 | >>>     go build \
  44 | >>>       -mod=readonly \
  45 | >>>       -tags "netgo,ledger,muslc" \
  46 | >>>       -ldflags "-X github.com/cosmos/cosmos-sdk/version.Name="neutron" \
  47 | >>>               -X github.com/cosmos/cosmos-sdk/version.AppName="neutrond" \
  48 | >>>               -X github.com/cosmos/cosmos-sdk/version.Version=${GIT_VERSION} \
  49 | >>>               -X github.com/cosmos/cosmos-sdk/version.Commit=${GIT_COMMIT} \
  50 | >>>               -X github.com/cosmos/cosmos-sdk/version.BuildTags='${BUILD_TAGS}' \
  51 | >>>               -X github.com/neutron-org/neutron/app.EnableSpecificProposals=${ENABLED_PROPOSALS} \
  52 | >>>               -w -s -linkmode=external -extldflags '-Wl,-z,muldefs -static'" \
  53 | >>>       -trimpath \
  54 | >>>       -o /neutron/build/neutrond \
  55 | >>>       /neutron/cmd/neutrond
  56 |
--------------------
ERROR: failed to solve: process "/bin/sh -c go build       -mod=readonly       -tags \"netgo,ledger,muslc\"       -ldflags \"-X github.com/cosmos/cosmos-sdk/version.Name=\"neutron\"               -X github.com/cosmos/cosmos-sdk/version.AppName=\"neutrond\"               -X github.com/cosmos/cosmos-sdk/version.Version=${GIT_VERSION}               -X github.com/cosmos/cosmos-sdk/version.Commit=${GIT_COMMIT}               -X github.com/cosmos/cosmos-sdk/version.BuildTags='${BUILD_TAGS}'               -X github.com/neutron-org/neutron/app.EnableSpecificProposals=${ENABLED_PROPOSALS}               -w -s -linkmode=external -extldflags '-Wl,-z,muldefs -static'\"       -trimpath       -o /neutron/build/neutrond       /neutron/cmd/neutrond" did not complete successfully: exit code: 1

View build details: docker-desktop://dashboard/build/neutronbuilder/neutronbuilder0/meusugaqygncuvs30uikwh8f0
make: *** [build-static-linux-amd64] Error 1
```

File: app/app.go [hunk 1]
```diff
@@ -711,7 +711,7 @@ func New(
 	// NOTE: we need staking feature here even if there is no staking module anymore because cosmwasm-std in the CosmWasm SDK requires this feature
 	// NOTE: cosmwasm_1_2 feature enables GovMsg::VoteWeighted, which doesn't work with Neutron, because it uses its own custom governance,
 	//       however, cosmwasm_1_2 also enables WasmMsg::Instantiate2, which works as one could expect
-	supportedFeatures := "iterator,stargate,staking,neutron,cosmwasm_1_1,cosmwasm_1_2,cosmwasm_1_3,cosmwasm_1_4"
+	supportedFeatures := []string{"iterator", "stargate", "staking", "neutron", "cosmwasm_1_1", "cosmwasm_1_2", "cosmwasm_1_3", "cosmwasm_1_4"}
```
Comment (mmikev) [inline app/app.go]: We need to enable: `cosmwasm_2_0`

**Review by pr0n00gler — CHANGES_REQUESTED** [review]
`Stargate` queries are deprecated in wasmvm 2.0 and there are `Grpc` queries now.
Please whitelist all the necessary `Grpc` queries as well + mark `Stargate` queries whitelisting as deprecated now

**Review by pr0n00gler — APPROVED** [review]

**Comment by pr0n00gler** [comment]
We must forget to move all our dev-contract and Neutron SDK to gRPC instead of Stargate.
Otherwise LGTM

**Pull request merged**